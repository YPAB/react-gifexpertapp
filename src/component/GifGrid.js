import React from "react";
import PropTypes from 'prop-types';
import { useFetchGifs } from "../hooks/useFetchGifs";
//import { getGifs } from "../helpers/getGifs";
import { GifGridItem } from "./GifGridItem";

export const GifGrid = ({category}) => {

    // const [images,setImages] = useState ([]);

    // es para que se renderice una sola vez (Solo se ejecuta cuando el componente es iniciado por primera vez)
   /*  useEffect(() => {
        getGifs(category)
        .then(setImages);
    }, [category])
*/
    //getGifs();

    const {data:images,loading} = useFetchGifs(category);

    return (
        <>

            <h3>{category}</h3>

            <div className='card-grid'>
                {loading && <p>Cargando ...</p>}

                
                    {
                        images.map (img => (

                            <GifGridItem 
                            key={img.id}
                            {...img}/>

                            
                        ))
                    }
                
            </div>
        </>
    )

}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}