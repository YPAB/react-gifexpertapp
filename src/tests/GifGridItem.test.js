import React from "react";
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { GifGridItem } from "../component/GifGridItem";

describe ('Pruebas en GigGridItem',() => {

    const id = 2;
    const title = 'Homero';
    const url = 'https://media0.giphy.com/media/Zk9mW5OmXTz9e/giphy.gif?cid=3271b137w2s82xs8dgbbvuyi6y9qizhmmp3vs5cx7vpl3jm2&rid=giphy.gif&ct=g'

    const wrapper = shallow(<GifGridItem id= {id} title={title} url={url}/>)

    test ('debe devolver el objeto GifGridItem',()=> {
        

        expect(wrapper).toMatchSnapshot();
    });

    test ('debe de terner un parrafo con el titulo', () => {

        const p = wrapper.find('p');

        expect( p.text().trim() ).toBe( title );

    });


    test('debe de tener una imagen igual al url y alt de los props',() => {
        
        const img  = wrapper.find('img');

        // img.props => tenemos acceso a un objeto de las propiedades que tiene la imagen
        // img.prop('src') => se puede acceder a un atributo.
        
        expect( img.prop('src') ).toBe(url);
        expect ( img.prop('alt') ).toBe(title);
    });

    test('debe de tener animate__bounce',() => {
        const div = wrapper.find('div');

        expect( div.prop('className') ).toContain('animate__bounce');


    });



});