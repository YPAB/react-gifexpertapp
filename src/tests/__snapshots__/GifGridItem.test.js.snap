// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GigGridItem debe devolver el objeto GifGridItem 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="Homero"
    src="https://media0.giphy.com/media/Zk9mW5OmXTz9e/giphy.gif?cid=3271b137w2s82xs8dgbbvuyi6y9qizhmmp3vs5cx7vpl3jm2&rid=giphy.gif&ct=g"
  />
  <p>
    Homero
     
  </p>
</div>
`;
